sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.h

```diff
@@ -859,6 +859,16 @@ static inline void _evtimer_add(void *ctx, int16_t type,
     evtimer_add_msg(&_nib_evtimer, event, target_pid);
 }
 
+/**
+ * @brief   Removes an event from the event timer
+ *
+ * @param[in] event Representation of the event.
+ */
+static inline void _evtimer_del(evtimer_msg_event_t *event)
+{
+    evtimer_del((evtimer_t *)(&_nib_evtimer), (evtimer_event_t *)event);
```

Comment:
The casts should not be needed IMO. This at least compiles fine:

```diff
diff --git a/sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.h b/sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.h
index 76052e83d7..77ffb8f632 100644
--- a/sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.h
+++ b/sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.h
@@ -23,13 +23,13 @@
 #include <stdbool.h>
 #include <stdint.h>
 #include <string.h>
-#include <kernel_defines.h>
 
 #include "bitfield.h"
 #include "evtimer_msg.h"
 #include "sched.h"
 #include "mutex.h"
 #include "net/eui64.h"
+#include "kernel_defines.h"
 #include "net/ipv6/addr.h"
 #ifdef MODULE_GNRC_IPV6
 #include "net/gnrc/ipv6.h"
@@ -835,6 +835,16 @@ int _nib_get_route(const ipv6_addr_t *dst, gnrc_pktsnip_t *ctx,
  */
 uint32_t _evtimer_lookup(const void *ctx, uint16_t type);
 
+/**
+ * @brief   Removes an event from the event timer
+ *
+ * @param[in] event Representation of the event.
+ */
+static inline void _evtimer_del(evtimer_msg_event_t *event)
+{
+    evtimer_del(&_nib_evtimer, &event->event);
+}
+
 /**
  * @brief   Adds an event to the event timer
  *
@@ -851,7 +861,7 @@ static inline void _evtimer_add(void *ctx, int16_t type,
 #else
     kernel_pid_t target_pid = KERNEL_PID_LAST;  /* just for testing */
 #endif
-    evtimer_del((evtimer_t *)(&_nib_evtimer), (evtimer_event_t *)event);
+    _evtimer_del(event);
     event->event.next = NULL;
     event->event.offset = offset;
     event->msg.type = type;
@@ -859,16 +869,6 @@ static inline void _evtimer_add(void *ctx, int16_t type,
     evtimer_add_msg(&_nib_evtimer, event, target_pid);
 }
 
-/**
- * @brief   Removes an event from the event timer
- *
- * @param[in] event Representation of the event.
- */
-static inline void _evtimer_del(evtimer_msg_event_t *event)
-{
-    evtimer_del((evtimer_t *)(&_nib_evtimer), (evtimer_event_t *)event);
-}
-
 #ifdef __cplusplus
 }
 #endif
```

Reply:
(The change with `<kernel_defines.h>` vs `"kernel_defines.h" is correct but unrelated.)